docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+A common symptom of this issue is self-replicating lists, i.e.:
+
+```js
+// where model will = {someValue: [1], valueToUpdate: 5}
+const [model, handleUpdateModel] = useContext(UiSdkContext);
+// Create a shallow copy instead of using `lodash` or other method of deepcopy
+a = model;
+
+a.valueToUpdate = 8;
+// then use `handleUpdateModel` call
+// ... promise resovles and..
+>> 'a: {someValue: [1, 1], valueToUpdate: 8}'
+```
+
+Overall, this simple reusable component allows text input field values to be stored and updated via a given key and value within our config.
+You could add other inputs and fields like `FileInput` in this module.
+
+#### Component(s): `DataInfo.tsx`, `DataInputSection.tsx`
+
+##### `DataInfo.tsx`
+
+```jsx
+import React, { useContext } from 'react';
+import { Grid, Container, Typography, Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from '@alteryx/ui';
+import { Context as UiSdkContext } from '@alteryx/react-comms';
+import TokenTranslation from "./TokenTranslation";
+
+const DataTable = ({rows}) => {
+
+  return (
+  <TableContainer component={Paper}>
+    <Table size="small">
+      <TableHead>
+        <TableRow>
+          <TableCell>Label </TableCell>
+          <TableCell align="left">Review</TableCell>
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        {rows.map((row, rowNum) => (
+          <TableRow key={`row-${rowNum}}`}>
+            <TableCell component="th" scope="row">
+              {row.label}
+            </TableCell>
+            <TableCell align="right">{row.text}</TableCell>
+          </TableRow>
+        ))}
+      </TableBody>
+    </Table>
+  </TableContainer>)
+};
+
+const VectorizationPreview = ({sampleRaw, sampleVectorized}) => {
+    return (<Grid spacing={1} container>
+        <Grid item> {sampleRaw} </Grid>
+        <Grid item> {sampleVectorized} </Grid>
+    </Grid>)
+}
+
+export const DataInfo = () => {
+    const [model] = useContext(UiSdkContext)
+    const tableRows = model.Configuration.datasetInfo.rawTrainSample.textBatch.map((val, index) => {
+        return {text: val, label: model.Configuration.datasetInfo.rawTrainSample.labelBatch[index]}
+    })
+    let vectorzationSample = model.Configuration.datasetInfo.vectorizedSample
+    return (
+        <Container>
+        <Typography variant="h1"> Data Info </Typography>
+        <Grid alignItems="flex-end" container spacing={1}>
+            <Grid item> <DataTable rows={tableRows} /> </Grid>
+            <Grid item> <VectorizationPreview {...vectorzationSample} /> </Grid>
+            <Grid item> <TokenTranslation /></Grid>
+        </Grid>
+        </Container>)
+}
+```
+
+A data table that accepts `tableRows` to preview samples of our test data as it feeds into our model. Additionally, some vectorization sampling and token samples, as we've defined them in our backend.
```

Comment:
TW review:

A data table that accepts `tableRows` to preview samples of the test data as it feeds into the model. Additionally, some vectorization sampling and token samples, as we've defined them in the back end.